Plugin~/WebRTCPlugin/GraphicsDevice/OpenGL/OpenGLContext.cpp

```diff
@@ -89,68 +107,81 @@ namespace webrtc
             };
 
             int returnedElements;
-            GLXFBConfig* configs = glXChooseFBConfig(display_, 0, visualAttrs, &returnedElements);
-            pbuffer_ = glXCreatePbuffer(display_, configs[0], attrs);
+            GLXFBConfig* configs = glXChooseFBConfig(display, 0, visualAttrs, &returnedElements);
+            pbuffer_ = glXCreatePbuffer(display, configs[0], attrs);
             RTC_DCHECK(pbuffer_);
-            if(!glXMakeContextCurrent(display_, pbuffer_, pbuffer_, context_))
+            if(!glXMakeContextCurrent(display, pbuffer_, pbuffer_, context_))
             {
                 RTC_LOG(LS_ERROR) << "glXMakeContextCurrent failed";
                 throw;
             }
+            XFree(configs);
+            configs = nullptr;
+            created_ = true;
         }
         ~GLXContextImpl()
         {
-            glXDestroyPbuffer(display_, pbuffer_);
-            glXDestroyContext(display_, context_);
+            glXDestroyPbuffer(display, pbuffer_);
+            glXDestroyContext(display, context_);
         }
+
+        GLXContext context() const { return context_; }
+
+        static Display* display;
     private:
         GLXPbuffer pbuffer_;
         GLXContext context_;
-        Display* display_;
+        // In Unity, this flag is false because the context already created which rely on the render thread.
+        bool created_;
     };
+    Display* GLXContextImpl::display;
 
-    // The context is created by Unity, and it related on the render thread.
-    GLXContext g_context;
-    Display* g_display = nullptr;
 #endif
 
-    void OpenGLContext::InitGLContext()
+    void OpenGLContext::Init()
     {
 #if SUPPORT_OPENGL_ES
-        g_context = eglGetCurrentContext();
-        RTC_DCHECK(g_context);
-        g_display = eglGetCurrentDisplay();
-        RTC_DCHECK(g_display);
+        EGLContextImpl::display = eglGetDisplay(EGL_DEFAULT_DISPLAY);
+        RTC_DCHECK(EGLContextImpl::display);
+        EGLint major, minor;
+        if(!eglInitialize(EGLContextImpl::display, &major, &minor))
+        {
+            RTC_LOG(LS_ERROR) << "eglInitialize failed:" << eglGetError();
+            throw;
+        }
 #elif SUPPORT_OPENGL_CORE && UNITY_LINUX
-        g_context = glXGetCurrentContext();
-        RTC_DCHECK(g_context);
-        g_display = glXGetCurrentDisplay();
-        RTC_DCHECK(g_display);
+        GLXContextImpl::display = glXGetCurrentDisplay();
+        RTC_DCHECK(GLXContextImpl::display);
 #endif
     }
 
-    bool OpenGLContext::CurrentContext()
+    std::unique_ptr<OpenGLContext> OpenGLContext::CurrentContext()
     {
 #if SUPPORT_OPENGL_ES
-        return eglGetCurrentContext() != EGL_NO_CONTEXT;
-#elif SUPPORT_OPENGL_CORE && UNITY_LINUX
-        return glXGetCurrentContext() != NULL;
+        if(eglGetCurrentContext())
+            return CreateGLContext();
```

Comment:
I think this method needs argument.

```suggestion
            return CreateGLContext(eglGetCurrentContext());
```